### PR TITLE
Fix python linter

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
- Fixes flake8 exclude files
- Fixes hack/python_releash.sh script 
- Adds black as dev dependency

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/python-test.yml [hunk 1]
```diff
@@ -34,8 +34,9 @@ jobs:
       - name: Check formatting with black
         uses: psf/black@stable
         with:
-          options: "--check"
+          options: "--check --diff"
           src: "python"
```
Comment (sivanantha321) [inline .github/workflows/python-test.yml]: Should we lint files under test/e2e as well ?

Comment (yuzisun) [reply .github/workflows/python-test.yml]: Are we not linting them ?

Comment (sivanantha321) [reply .github/workflows/python-test.yml]: Yep, because we provided source as python folder

Comment (yuzisun) [reply .github/workflows/python-test.yml]: I see, we should lint all our python code.

Comment (cmaddalozzo) [reply .github/workflows/python-test.yml]: Makes sense to me. @sivanantha321 can you run `black` on the `test/e2e` folder so the code is properly formatted?

Comment (sivanantha321) [reply .github/workflows/python-test.yml]: Done

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Comment by sivanantha321** [comment]
Black does not seem to oragnize the imports. Should we use [isort](https://pycqa.github.io/isort/docs/configuration/black_compatibility.html) kind of tool for that ?

**Pull request merged**